Description: Make GAPI optional
 GAPI is only needed for the google resource. If GAPI is not available simply skip
 building this resource.
Author: Hefee <hefee@debian.org>
Origin: Debian
Bug-Debian: https://bugs.debian.org/925155
Last-Update: 2019-03-20
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/resources/CMakeLists.txt
+++ b/resources/CMakeLists.txt
@@ -52,7 +52,9 @@ add_subdirectory( maildir )
 add_subdirectory( openxchange )
 add_subdirectory( pop3 )
 
-add_subdirectory( google-groupware )
+if (KPimGAPI_FOUND)
+  add_subdirectory( google-groupware )
+endif()
 
 add_subdirectory( shared )
 add_subdirectory( birthdays )
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,7 +135,7 @@ find_package(KF5IMAP ${KIMAP_LIB_VERSION
 find_package(KF5AkonadiNotes ${AKONADINOTE_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5AkonadiCalendar ${AKONADICALENDAR_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5PimCommon ${PIMCOMMON_LIB_VERSION} CONFIG REQUIRED)
-find_package(KPimGAPI ${KGAPI_LIB_VERSION} CONFIG REQUIRED)
+find_package(KPimGAPI ${KGAPI_LIB_VERSION} CONFIG)
 find_package(KF5Libkdepim ${LIBKDEPIM_LIB_VERSION_LIB} CONFIG REQUIRED)
 find_package(KF5Ldap ${KLDAP_LIB_VERSION} CONFIG REQUIRED)
 
--- a/resources/imap/CMakeLists.txt
+++ b/resources/imap/CMakeLists.txt
@@ -1,6 +1,10 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"akonadi_imap_resource\")
 
-set(WITH_GMAIL_XOAUTH2 TRUE)
+if (KPimGAPI_FOUND)
+  set(WITH_GMAIL_XOAUTH2 TRUE)
+else()
+  set(WITH_GMAIL_XOAUTH2 FALSE)
+endif()
 
 configure_file(config-imap.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-imap.h)
 
@@ -91,6 +95,15 @@ target_link_libraries(imapresource
 )
 
 ########### next target ###############
+if (KPimGAPI_FOUND)
+   set(akonadi_imap_resource_KPimGAPI
+      KPim::GAPICore
+   )
+   set( akonadi_imap_resource_KPimGAPI_SRCS
+      passwordrequester.cpp
+      gmailpasswordrequester.cpp
+   )
+endif()
 
 set( akonadi_imap_resource_SRCS
   main.cpp
@@ -99,8 +112,7 @@ set( akonadi_imap_resource_SRCS
   settingspasswordrequester.cpp
   setupserver.cpp
   serverinfodialog.cpp
-  passwordrequester.cpp
-  gmailpasswordrequester.cpp
+  ${akonadi_imap_resource_KPimGAPI_SRCS}
 )
 
 install( FILES imapresource.desktop DESTINATION "${KDE_INSTALL_DATAROOTDIR}/akonadi/agents" )
@@ -130,9 +142,12 @@ target_link_libraries(akonadi_imap_resou
     imapresource
     KF5::TextWidgets
     folderarchivesettings
-    KPim::GAPICore
+    ${akonadi_imap_resource_KPimGAPI}
 )
 
+
+
+
 install(TARGETS akonadi_imap_resource ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
 add_subdirectory( wizard )
